pkgs/tools/misc/idutils/default.nix

```diff
@@ -1,27 +1,31 @@
-{ fetchurl, lib, stdenv, emacs }:
+{ fetchurl, lib, stdenv, emacs, gnulib, autoconf, bison, automake, gettext, gperf, texinfo, perl, rsync, gawk }:
 
 stdenv.mkDerivation rec {
-  name = "idutils-4.6";
+  pname = "idutils";
+  version = "4.6";
 
   src = fetchurl {
-    url = "mirror://gnu/idutils/${name}.tar.xz";
+    url = "mirror://gnu/idutils/${pname}-${version}.tar.xz";
     sha256 = "1hmai3422iaqnp34kkzxdnywl7n7pvlxp11vrw66ybxn9wxg90c1";
   };
 
   preConfigure =
     ''
-       # Fix for building on Glibc 2.16.  Won't be needed once the
-       # gnulib in idutils is updated.
-       sed -i '/gets is a security hole/d' lib/stdio.in.h
+    # replace embedded gnulib tests with those from gnulib package
+    shopt -s extglob
+    (cd gnulib-tests; rm -r !(Makefile.am))
+    shopt -u extglob
+    substituteInPlace ./configure.ac --replace "AC_PREREQ(2.61)" "AC_PREREQ(2.64)"
+    ./bootstrap --force --gnulib-srcdir=${gnulib} --skip-po --bootstrap-sync --no-git
     '';
 
-  buildInputs = lib.optional stdenv.isLinux emacs;
+  buildInputs = [ gnulib autoconf bison automake gettext gperf texinfo perl rsync gawk ] ++ lib.optional stdenv.isLinux emacs;
```

Comment:
The `autoconf`,  `authomake`, `bison`, `texinfo`, `perl`, `rsync` and `gawk` all likely should be in `nativeBuildInputs` unless they are used as runtime dependencies. Also I assume you remembered to try using `autoreconfHook`, right?

Reply:
I moved those to nativeBuildInputs.  Regarding autoreconfHook, idutils requires gnulib's bootstrap to copy parts of gnulib into the idutils build tree. Then, bootstrap invokes the autotools in a special order (see below). This process subsumes the autoreconHook:
`# NOTE: we have to be careful to run both autopoint and libtoolize
# before gnulib-tool, since gnulib-tool is likely to provide newer
# versions of files "installed" by these two programs.
# Then, *after* gnulib-tool (see below), we have to be careful to
# run autoreconf in such a way that it does not run either of these
# two just-pre-run programs.`